include/spdk_internal/bdev.h

```diff
@@ -271,6 +271,13 @@ struct spdk_bdev_io {
 		struct {
 			enum spdk_bdev_reset_type type;
 		} reset;
+		struct {
+			struct spdk_nvme_cmd cmd;
+			void *buf;
+			size_t nbytes;
+			uint32_t p_lcore;
```

Comment:
The lcore is used to pass a message back to the correct core, but the existence of the message pass is purely an implementation detail of the blockdev_nvme module. Instead, store the lcore in struct nvme_bdev_io.

Or, you can probably store a struct spdk_event * in the nvme_bdev_io that you allocate at submission time. That way if the event allocation fails, you can immediately fail the command back to the user.